src/components/Store.tsx

```diff
@@ -189,9 +190,14 @@ function reducer(state: ChatState, action: A): ChatState {
                 session: {
                     wallet: new Wallet(session.privateKey),
                     streamrClient,
-                    provider: provider,
                 },
             }
+        case ActionType.SetEthereumProvider:
+            return {
+                ...state,
+                ethereumProvider: action.payload || undefined,
+                ethereumProviderReady: true,
```

Comment:
`ethereumProviderReady` will tell you whether we've tried detecting the provider or not (`false` if we haven't).

Adding it before we can use it in the UI but we're gonna need it to inform the user that their browser isn't ready for the revolution! And since we don't want to show warnings prematurely (before we checked the revolution-readyness) we got this flag. Trust me, it's gon' be good.